src/common/io/read.intern.h

```diff
@@ -15,6 +15,7 @@ Constructors
 typedef struct IoReadInterface
 {
     bool block;                                               // Do reads block when buffer is larger than available bytes?
+    size_t totalBytesRead;                                    // Total bytes read
```

Comment:
This should be in the interface object (https://github.com/pgbackrest/pgbackrest/blob/release/2.48/src/common/io/read.c#L17) and exposed via a getter function, e.g. `ioReadTotal()`.

Reply:
I did exactly this in one of the previous versions of the patch.

Reply:
> This should be in the interface object

in IoReadInterface or in IoRead?

Reply:
In IoRead, here: https://github.com/pgbackrest/pgbackrest/blob/release/2.48/src/common/io/read.c#L17

Reply:
done